### DISCUSSION TITLE
[proposal] improve inference when passing methods as generic func parameter

### DISCUSSION THREAD

**Discussion by leandromoh** [dotnet/csharplang#3722] [Language Ideas]
lets consider follow methods:

```c#
string Operation(int a);

void Foo(Func<int, string> func);

void Bar<T1, TResult>(Func<T1, TResult> func);

Foo(Operation); // this compiles!
Bar(Operation); // this not
```

Is there some way to improve this inference?

**Comment by CyrusNajmabadi** [comment]
> Is there some way to improve this inference?

Do you have a language proposal on how to improve the inference here?

**Comment by JustNrik** [comment]
`Bar<int, string>(Operation)` will work, but I don't know why the compiler can't infer it, it really should be able to infer it... I think?

**Comment by jnm2** [comment]
@JustNrik A proposal would explain specifically how the inference is done, fitting the steps it would take to break down the problem into the existing spec.

**Comment by leandromoh** [comment]
@cston does the "lambda improviments" feature resolves this kind of inference?

**Reply by 333fred** [reply]
Yes.

**Reply by leandromoh** [reply]
awesome! thanks

**Reply by leandromoh** [reply]
the follow code will  be valid too?

```c#
string Operation(int a);

var f = Operation; // infer that f has type Func<int, string>
```

**Reply by 333fred** [reply]
Yes.
